zeus/metric.py

```diff
@@ -288,28 +319,36 @@ def begin_window(self, name: str) -> None:
                 self.update_period,
                 self.prometheus_url,
                 self.job,
+                sync_execution,
```

Comment:
This is wrong. If `sync_execution` is `True`, you need to call `zeus.utils.framework.sync_execution` on the main thread there the application is running. The monitor process's `ZeusMonitor` should always be called with `sync_execution=False`.